### PR TITLE
feat(ts): stack and queue implementation

### PR DESCRIPTION
**Pull request by Progyan1997**

### CODE REVIEW

**Comment by leios** [comment]
thanks for the implementation; however, we do not yet have a chapter *with code* for stacks and queues. This will be updated soon, so I'll keep this PR in mind as we write the text for that section #771 has a similar issue, so hopefully we can work everything out!

File: contents/stacks_and_queues/code/typescript/queue.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+   * `size` return size or length of the queue
+   */
+  size(): number;
+  /**
+   * `front` returns first element of the queue
+   */
+  front(): T;
+}
+
+class Queue<T> implements IQueue<T> {
```
Comment (ntindle) [inline contents/stacks_and_queues/code/typescript/queue.ts:20]: @leios do we have standards for TypeHinting in the interface names?

**Review by ntindle — COMMENTED** [review]
Am typescript dev, this LGTM

File: contents/stacks_and_queues/code/typescript/queue.ts (hunk 1 above)
Comment (leios) [reply contents/stacks_and_queues/code/typescript/queue.ts:20]: I don't think so. Maybe we should add something to the code style guide?

**Review by leios — APPROVED** [review]
If it looks good to @ntindle, then I am happy to approve as well!

**Comment by leios** [comment]
I will go ahead and merge this now and then put the infrastructure in another PR to solve issue #838

**Pull request merged**